tests/rewards/test_reward_nets.py

```diff
@@ -165,6 +245,55 @@ def test_reward_valid(env_name, reward_type, tmpdir):
     assert isinstance(pred_reward[0], numbers.Number)
 
 
+@pytest.mark.parametrize("env_name", IMAGE_ENVS)
+@pytest.mark.parametrize("reward_type", DESERIALIZATION_TYPES)
+def test_reward_valid_image(env_name, reward_type, tmpdir):
+    """Test output of reward function is appropriate shape and type."""
+    venv = util.make_vec_env(env_name, n_envs=1, parallel=False)
+    venv, tmppath = _make_env_and_save_reward_net(
+        env_name,
+        reward_type,
+        tmpdir,
+        is_image=True,
+    )
+
+    TRAJECTORY_LEN = 10
+    obs = _sample(venv.observation_space, TRAJECTORY_LEN)
+    actions = _sample(venv.action_space, TRAJECTORY_LEN)
+    next_obs = _sample(venv.observation_space, TRAJECTORY_LEN)
+    steps = np.arange(0, TRAJECTORY_LEN)
+
+    reward_fn = serialize.load_reward(reward_type, tmppath, venv)
+    pred_reward = reward_fn(obs, actions, next_obs, steps)
+
+    assert pred_reward.shape == (TRAJECTORY_LEN,)
+    assert isinstance(pred_reward[0], numbers.Number)
```

Comment:
We seem to be assuming `pred_reward` is an `np.ndarray`, so maybe we should just check the dtype? Note in principle it's possible that `pred_reward[0]` is numeric whereas `pred_reward[1]` is not if `pred_reward` were an object array (it'd be *weird*... but allowed.)

I think `np.issubdtype(pred_reward.dtype, np.number)` would do the trick.